gcloud-java-core/src/main/java/com/google/gcloud/PageImpl.java

```diff
@@ -35,6 +37,50 @@
     Page<T> nextPage();
   }
 
+  static class PageIterator<T> implements java.util.Iterator<T> {
```

Comment:
Could be simplified by extending  [AbstractIterator](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/AbstractIterator.html)